best-practices/tidb-partitioned-tables-best-practices.md

```diff
@@ -0,0 +1,691 @@
+
+#### Types of tables to be tested
+
+The query performance of the following types of tables are evaluated:
+
+- Non-partitioned tables
+- Partitioned tables with global indexes
+- Partitioned tables with local indexes
+
+#### Test setup
+
+- The partitioned table has 365 partitions, defined by the range partitioning on a date column.
+- Each matching key returns multiple rows, simulating a high-volume OLTP-style query pattern.
+- The impact of different partition counts is also evaluated to understand how partition granularity influences latency and index performance.
+
+#### Schema
+
+The following schema is used in the example.
+
+```sql
+CREATE TABLE `fa` (
+  `id` bigint NOT NULL AUTO_INCREMENT,
+  `account_id` bigint(20) NOT NULL,
+  `sid` bigint(20) DEFAULT NULL,
+  `user_id` bigint NOT NULL,
+  `date` int NOT NULL,
+  PRIMARY KEY (`id`,`date`) /*T![clustered_index] CLUSTERED */,
+  KEY `index_fa_on_sid` (`sid`),
+  KEY `index_fa_on_account_id` (`account_id`),
+  KEY `index_fa_on_user_id` (`user_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+PARTITION BY RANGE (`date`)
+(PARTITION `fa_2024001` VALUES LESS THAN (2025001),
+PARTITION `fa_2024002` VALUES LESS THAN (2025002),
+PARTITION `fa_2024003` VALUES LESS THAN (2025003),
+...
+...
+PARTITION `fa_2024365` VALUES LESS THAN (2025365));
+```
+
+#### SQL
+
+The following SQL statement is used in the example.
+
+```sql
+SELECT `fa`.*
+FROM `fa`
+WHERE `fa`.`sid` IN (
+  1696271179344,
+  1696317134004,
+  1696181972136,
+  ...
+  1696159221765
+);
+```
+
+- Query filters on secondary index, but does not include the partition key.
+- Causes local indexes key lookup for each partition due to lack of pruning.
+- Table lookup tasks are significantly higher for partitioned tables.
+
+#### Test results
+
+The following table shows the test results.
+
+| Configuration | Average Query Time | Cop task for index range scan | Cop task for table lookup | Total Cop tasks | Key Takeaways |
+|---|---|---|---|---|---|
+| Non-partitioned table | 12.6 ms | 72 | 79 | 151 | Provides the best performance with the fewest Cop tasks, which is ideal for most OLTP use cases. |
+| Partitioned table with local indexes | 108 ms | 600 | 375 | 975 | When the partition key is not used in the query condition, local index queries scan all partitions. |
+| Partitioned table with global indexes | 14.8 ms | 69 | 383 | 452 | It improves index scan efficiency, but table lookups can still take a long time if many rows match. |
+
+Data comes from a table with 365 range partitions (for example, by date).
+
+- The **Average Query Time** is obtained from the `statement_summary` view.
+- The query uses a secondary index and returns 400 rows.
```

Comment:
```suggestion
```
Consolidate the context into a single introductory sentence in L110. The current format separates related information across a paragraph and bullet points, which is harder to scan.